ecs-cli/modules/cli/compose/project/project.go

```diff
@@ -174,6 +180,19 @@ func (p *ecsProject) parseECSParams() error {
 	return nil
 }
 
+func (p *ecsProject) parsePrivateRegistryContext() error {
+	logrus.Debug("Parsing the ecs-registry-creds yaml...")
+	registryCredsFileName := p.ecsContext.CLIContext.GlobalString(flags.RegistryCredsFileNameFlag)
```

Comment:
Be careful when using `CLIContext.GlobalStrin`g- it doesn't actually do what the name describes. In `urfave/cli`, each command gets a context and this is chained. `GlobalString` just looks one level up the context. So if we do "ecs-cli compose  create" the context is:

1. "ecs-cli"
2. "compose"
3. "create"

So `GlobalString` would look at "compose". However, if you have "ecs-cli compose service up" it should be looking at "service" since that is one level up from the bottom of the context. This is why I built and recommend using the recursive flag search function that is used in command_config.go

Reply:
👍  oh good to know, thanks!